### PR TITLE
feat(std/wasi): add start method to Context

### PR DESCRIPTION
**Pull request by caspervonb**
This adds a start method to the Context to make starting a command less tedious and yield consistent errors.

Manually setting the memory is still valid for more complex scenarios, just undocumented for the time being (haven't decided yet if it will be public or not).

### CODE REVIEW

File: std/wasi/snapshot_preview1.ts [hunk 1]
```diff
@@ -1556,4 +1555,41 @@ export default class Context {
...
+      throw new TypeError("WebAsembly.instance must provide a memory export");
+    }
+
+    this.memory = memory;
+
+    if (typeof _initialize !== "function" && _initialize !== undefined) {
+      throw new TypeError(
+        "WebAsembly.instance export _initialize must not be defined",
+      );
+    }
```
Comment (ry) [inline std/wasi/snapshot_preview1.ts:1585]: Not sure but shouldn't it be something like this?
```js
  if (!_initialize) {
    throw new TypeError(
      "WebAsembly.instance export _initialize must not be defined",
    );
  }
```

Comment (caspervonb) [reply std/wasi/snapshot_preview1.ts:1585]: Yikes.

Not quite but it is indeed wrong, should just be checking if _initialize is a function and throw if it is (there are two types of entry points, _initialize, or _start).

**Review by crypt096 — APPROVED** [review]

File: std/wasi/snapshot_preview1.ts [hunk 2]
```diff
@@ -285,7 +284,7 @@ export default class Context {
   constructor(options: ContextOptions) {
     this.args = options.args ? options.args : [];
     this.env = options.env ? options.env : {};
-    this.memory = options.memory!;
+    this.memory = null!;
```
Comment (bartlomieju) [inline std/wasi/snapshot_preview1.ts:287]: `!` is not needed here?

Comment (caspervonb) [reply std/wasi/snapshot_preview1.ts:287]: >  memory: WebAssembly.Memory;		

It's actually nullable according to the type rules, it's forced to null here and assigned/validated later in the start method.

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**